sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -83,22 +83,28 @@ def plot(self, include_values=True, cmap='viridis',
         n_classes = cm.shape[0]
         self.im_ = ax.imshow(cm, interpolation='nearest', cmap=cmap)
         self.text_ = None
-
         cmap_min, cmap_max = self.im_.cmap(0), self.im_.cmap(256)
 
         if include_values:
             self.text_ = np.empty_like(cm, dtype=object)
-            if values_format is None:
-                values_format = '.2g'
 
             # print text with appropriate color depending on background
             thresh = (cm.max() + cm.min()) / 2.0
+
             for i, j in product(range(n_classes), range(n_classes)):
                 color = cmap_max if cm[i, j] < thresh else cmap_min
-                self.text_[i, j] = ax.text(j, i,
-                                           format(cm[i, j], values_format),
-                                           ha="center", va="center",
-                                           color=color)
+                fmt = values_format
+
+                if values_format is None:
+                    if cm.dtype.kind == "f" or cm[i, j] >= 1e7:
```

Comment:
If the matrix is normalised, won't its values be <1? Should we be showing normalized values as %?

Reply:
Yes that's true <= 1 (in the case all predictions are correct). Wouldn't it be easier to let this be specified by the user in the case of normalization? The edge cases to choose if % or 'd'?

Also, the way plot_confusion_matrix is defined now is by letting users pass in an estimator, X and y_true. 

Wouldn't it make more sense to use y_pred and y_true and pass those as arguments?

Reply:
at this point it might be easier to literally check the length of the formatted string with both options and just use the shortest one

Reply:
@NicolasHug The format will then go to '.2g' at values => 100 (which is the same as before)

Reply:
@NicolasHug Values will then change to '.2g' format once >= 1e6

Reply:
Currently this does not always pick the shorter one. if `cm[i, j] == int(1e6)`, `fmt='d'` will be used, which results in: `1000000`, while `fmt='.2g'` gives `'1e+06'`